2025-05-24 12:12:11,023 | Level: ERROR | Module: views | Line: 1565 | Message: Role is required.
2025-05-24 12:15:39,408 | Level: ERROR | Module: views | Line: 1565 | Message: Role is required.
2025-05-24 12:15:39,411 | Level: ERROR | Module: views | Line: 1570 | Message: Email is required.
2025-05-24 12:15:39,412 | Level: ERROR | Module: views | Line: 1611 | Message: DOJ is required.
2025-05-24 12:15:39,413 | Level: ERROR | Module: views | Line: 1630 | Message: Department is required.
2025-05-24 12:15:39,413 | Level: ERROR | Module: views | Line: 1639 | Message: Designation is required.
2025-05-24 12:16:26,717 | Level: ERROR | Module: views | Line: 1565 | Message: Role is required.
2025-05-24 12:16:26,718 | Level: ERROR | Module: views | Line: 1570 | Message: Email is required.
2025-05-24 12:16:26,719 | Level: ERROR | Module: views | Line: 1611 | Message: DOJ is required.
2025-05-24 12:16:26,719 | Level: ERROR | Module: views | Line: 1630 | Message: Department is required.
2025-05-24 12:16:26,720 | Level: ERROR | Module: views | Line: 1639 | Message: Designation is required.
2025-05-24 15:27:27,581 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /assets/add/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 3853, in add_asset
    return render(request, 'assets_templates/Assets_add.html', {'employees': employees})
2025-05-26 14:06:32,764 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /error-log/ HTTP/1.1" 500 80836
2025-05-26 14:10:55,678 | Level: ERROR | Module: views | Line: 1763 | Message: Incorrect password.
2025-05-26 14:10:56,359 | Level: ERROR | Module: views | Line: 1763 | Message: Incorrect password.
2025-05-26 14:11:07,138 | Level: ERROR | Module: views | Line: 1759 | Message: No account found with this email.
2025-05-26 14:27:55,026 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /assets/list/

2025-05-27 14:30:07,987 | Level: ERROR | Module: views | Line: 1781 | Message: Incorrect password.
2025-05-27 14:30:18,291 | Level: ERROR | Module: views | Line: 1781 | Message: Incorrect password.
2025-05-27 14:30:21,804 | Level: ERROR | Module: views | Line: 1781 | Message: Incorrect password.
2025-05-27 15:50:34,109 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /delete/64/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 1325, in delete_employee
    logger.info(
    ~~~~~~~~~~~^
        "Employee Inactive",
        ^^^^^^^^^^^^^^^^^^^^
        extra={'action': 'Employee Inactive', 'message': f"Employee '{employee.name}' with ID {employee.id}."}
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\acer\AppData\Local\Programs\Python\Python313\Lib\logging\__init__.py", line 1519, in info
    self._log(INFO, msg, args, **kwargs)
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\acer\AppData\Local\Programs\Python\Python313\Lib\logging\__init__.py", line 1662, in _log
    record = self.makeRecord(self.name, level, fn, lno, msg, args,
                             exc_info, func, extra, sinfo)
  File "C:\Users\acer\AppData\Local\Programs\Python\Python313\Lib\logging\__init__.py", line 1636, in makeRecord
    raise KeyError("Attempt to overwrite %r in LogRecord" % key)
KeyError: "Attempt to overwrite 'message' in LogRecord"
2025-05-27 15:50:34,117 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /delete/64/ HTTP/1.1" 500 84079
2025-05-27 15:51:20,582 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /delete/64/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 1325, in delete_employee
    logger.info(
    ~~~~~~~~~~~^
        "Employee Inactive",
        ^^^^^^^^^^^^^^^^^^^^
        extra={'action': 'Employee Inactive', 'message': f"Employee '{employee.name}' with ID {employee.id}."}
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\acer\AppData\Local\Programs\Python\Python313\Lib\logging\__init__.py", line 1519, in info
    self._log(INFO, msg, args, **kwargs)
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\acer\AppData\Local\Programs\Python\Python313\Lib\logging\__init__.py", line 1662, in _log
    record = self.makeRecord(self.name, level, fn, lno, msg, args,
                             exc_info, func, extra, sinfo)
  File "C:\Users\acer\AppData\Local\Programs\Python\Python313\Lib\logging\__init__.py", line 1636, in makeRecord
    raise KeyError("Attempt to overwrite %r in LogRecord" % key)
KeyError: "Attempt to overwrite 'message' in LogRecord"
2025-05-27 15:51:20,591 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /delete/64/ HTTP/1.1" 500 84036
2025-05-27 15:52:22,140 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /delete/64/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 1325, in delete_employee
    logger.info(
    ~~~~~~~~~~~^
        "Employee Inactive",
        ^^^^^^^^^^^^^^^^^^^^
        extra={'action': 'Employee Inactive', 'message': f"Employee '{employee.name}' with ID {employee.id}."}
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\acer\AppData\Local\Programs\Python\Python313\Lib\logging\__init__.py", line 1519, in info
    self._log(INFO, msg, args, **kwargs)
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\acer\AppData\Local\Programs\Python\Python313\Lib\logging\__init__.py", line 1662, in _log
    record = self.makeRecord(self.name, level, fn, lno, msg, args,
                             exc_info, func, extra, sinfo)
  File "C:\Users\acer\AppData\Local\Programs\Python\Python313\Lib\logging\__init__.py", line 1636, in makeRecord
    raise KeyError("Attempt to overwrite %r in LogRecord" % key)
KeyError: "Attempt to overwrite 'message' in LogRecord"
2025-05-27 15:52:22,144 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /delete/64/ HTTP/1.1" 500 84036
2025-05-27 16:24:49,444 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /Login/user/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\utils\decorators.py", line 192, in _view_wrapper
    result = _process_exception(request, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\utils\decorators.py", line 190, in _view_wrapper
    response = view_func(request, *args, **kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 1814, in Login_user
    logger.error("No account found with this email.")
    ^^^^^^
UnboundLocalError: cannot access local variable 'logger' where it is not associated with a value
2025-05-27 16:24:49,453 | Level: ERROR | Module: basehttp | Line: 213 | Message: "POST /Login/user/ HTTP/1.1" 500 79701
2025-05-27 16:27:16,307 | Level: ERROR | Module: views | Line: 1837 | Message: Incorrect password.
2025-05-27 17:21:29,318 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /assets/add/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4012, in add_asset
    'details': f"Employee '{Asset.allocated_to.name}' with ID {Asset.allocated_to.id} assest was Added"
                            ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ForwardManyToOneDescriptor' object has no attribute 'name'
2025-05-27 17:21:29,323 | Level: ERROR | Module: basehttp | Line: 213 | Message: "POST /assets/add/ HTTP/1.1" 500 71496
2025-05-27 18:24:05,291 | Level: ERROR | Module: views | Line: 1814 | Message: No account found with this email.
2025-05-27 18:24:30,290 | Level: ERROR | Module: views | Line: 1814 | Message: No account found with this email.
2025-05-27 18:24:36,735 | Level: ERROR | Module: views | Line: 1814 | Message: No account found with this email.
2025-05-27 18:25:03,821 | Level: ERROR | Module: views | Line: 1844 | Message: Incorrect password.
2025-05-27 18:25:07,724 | Level: ERROR | Module: views | Line: 1844 | Message: Incorrect password.
2025-05-27 18:25:11,080 | Level: ERROR | Module: views | Line: 1844 | Message: Incorrect password.
2025-05-27 18:25:14,103 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /Login/user/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\utils\decorators.py", line 192, in _view_wrapper
    result = _process_exception(request, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\utils\decorators.py", line 190, in _view_wrapper
    response = view_func(request, *args, **kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 1820, in Login_user
    if user.last_failed_login and datetime.now() < user.last_failed_login + timedelta(hours=2):
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: can't compare offset-naive and offset-aware datetimes
2025-05-27 18:25:14,112 | Level: ERROR | Module: basehttp | Line: 213 | Message: "POST /Login/user/ HTTP/1.1" 500 79962
2025-05-27 18:25:43,705 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /Login/user/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\utils\decorators.py", line 192, in _view_wrapper
    result = _process_exception(request, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\utils\decorators.py", line 190, in _view_wrapper
    response = view_func(request, *args, **kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 1822, in Login_user
    remaining_time = lock_end_time - datetime.now()
                     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
TypeError: can't subtract offset-naive and offset-aware datetimes
2025-05-27 18:25:43,708 | Level: ERROR | Module: basehttp | Line: 213 | Message: "POST /Login/user/ HTTP/1.1" 500 80136
2025-05-28 15:10:47,666 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /exit_management/details/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 3311, in Resignation_details
    resignation=Email.resignation
                ^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'resignation'
2025-05-28 15:10:47,682 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /exit_management/details/ HTTP/1.1" 500 68511
2025-05-28 15:17:53,614 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /exit_management/details/Emp/37/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 3508, in Resign_emp
    employee = resignation.employee
               ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'employee'
2025-05-28 15:17:53,624 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /exit_management/details/Emp/37/ HTTP/1.1" 500 67784
2025-05-28 15:31:10,016 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /exit_management/details/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 3311, in Resignation_details
    resignation=Email.resignation
                ^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'resignation'
2025-05-28 15:31:10,023 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /exit_management/details/ HTTP/1.1" 500 68511
2025-05-28 15:31:16,682 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /exit_management/details/
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 3311, in Resignation_details
    resignation=Email.resignation
                ^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'resignation'
2025-05-28 15:31:16,685 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /exit_management/details/ HTTP/1.1" 500 68511
2025-05-28 18:27:49,113 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 510, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'break'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4238, in attendance_report
    return render(request, 'report_templates/Attendence.html', {
        'departments': departments,
    ...<7 lines>...
        'selected_department': selected_department,
    })
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
        contents,
    ...<2 lines>...
        self.engine,
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 1540, in do_with
    nodelist = parser.parse(("endwith",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 1540, in do_with
    nodelist = parser.parse(("endwith",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 512, in parse
    self.invalid_block_tag(token, command, parse_until)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 567, in invalid_block_tag
    raise self.error(
    ...<8 lines>...
    )
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 92: 'break', expected 'endwith'. Did you forget to register or load this tag?
2025-05-28 18:27:49,139 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report HTTP/1.1" 500 313440
2025-05-28 18:28:19,918 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 510, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'break'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4238, in attendance_report
    return render(request, 'report_templates/Attendence.html', {
        'departments': departments,
    ...<7 lines>...
        'selected_department': selected_department,
    })
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
        contents,
    ...<2 lines>...
        self.engine,
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 1540, in do_with
    nodelist = parser.parse(("endwith",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 1540, in do_with
    nodelist = parser.parse(("endwith",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 512, in parse
    self.invalid_block_tag(token, command, parse_until)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 567, in invalid_block_tag
    raise self.error(
    ...<8 lines>...
    )
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 92: 'break', expected 'endwith'. Did you forget to register or load this tag?
2025-05-28 18:28:19,927 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report HTTP/1.1" 500 313415
2025-05-28 18:28:20,812 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 510, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'break'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4238, in attendance_report
    return render(request, 'report_templates/Attendence.html', {
        'departments': departments,
    ...<7 lines>...
        'selected_department': selected_department,
    })
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
        contents,
    ...<2 lines>...
        self.engine,
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 1540, in do_with
    nodelist = parser.parse(("endwith",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 1540, in do_with
    nodelist = parser.parse(("endwith",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 512, in parse
    self.invalid_block_tag(token, command, parse_until)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 567, in invalid_block_tag
    raise self.error(
    ...<8 lines>...
    )
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 92: 'break', expected 'endwith'. Did you forget to register or load this tag?
2025-05-28 18:28:20,817 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report HTTP/1.1" 500 313415
2025-05-28 18:29:23,737 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 510, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'break_loop'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4238, in attendance_report
    return render(request, 'report_templates/Attendence.html', {
        'departments': departments,
    ...<7 lines>...
        'selected_department': selected_department,
    })
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
        contents,
    ...<2 lines>...
        self.engine,
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 1540, in do_with
    nodelist = parser.parse(("endwith",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 1540, in do_with
    nodelist = parser.parse(("endwith",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 512, in parse
    self.invalid_block_tag(token, command, parse_until)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 567, in invalid_block_tag
    raise self.error(
    ...<8 lines>...
    )
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 92: 'break_loop', expected 'endwith'. Did you forget to register or load this tag?
2025-05-28 18:29:23,743 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report HTTP/1.1" 500 318305
2025-05-28 18:31:17,791 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4244, in attendance_report
    return render(request, 'attendance_calendar.html', {
        'departments': departments,
    ...<7 lines>...
        'selected_department': selected_department,
    })
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: attendance_calendar.html
2025-05-28 18:31:17,794 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report HTTP/1.1" 500 102395
2025-05-28 18:32:03,370 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\fields\__init__.py", line 2128, in get_prep_value
    return int(value)
ValueError: invalid literal for int() with base 10: 'IT'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4224, in attendance_report
    employees = employees.filter(department_id=selected_department)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\query.py", line 1481, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\query.py", line 1499, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\query.py", line 1506, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\sql\query.py", line 1643, in add_q
    clause, _ = self._add_q(q_object, can_reuse)
                ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\sql\query.py", line 1675, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ~~~~~~~~~~~~~~~~~^
        child,
        ^^^^^^
    ...<7 lines>...
        update_join_types=update_join_types,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\sql\query.py", line 1585, in build_filter
    condition = self.build_lookup(lookups, col, value)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\sql\query.py", line 1412, in build_lookup
    lookup = lookup_class(lhs, rhs)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\lookups.py", line 38, in __init__
    self.rhs = self.get_prep_lookup()
               ~~~~~~~~~~~~~~~~~~~~^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\fields\related_lookups.py", line 112, in get_prep_lookup
    self.rhs = target_field.get_prep_value(self.rhs)
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\fields\__init__.py", line 2130, in get_prep_value
    raise e.__class__(
        "Field '%s' expected a number but got %r." % (self.name, value),
    ) from e
ValueError: Field 'id' expected a number but got 'IT'.
2025-05-28 18:32:03,401 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report?department=IT&month=3&year=2024 HTTP/1.1" 500 126946
2025-05-28 18:44:14,004 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4225, in attendance_report
    employees = EmployeeBISP.objects.filter(
        status='active',
        date_of_joining__lte=end_date  # Only employees who joined on/before month end
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\query.py", line 1481, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\query.py", line 1499, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\query.py", line 1506, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\sql\query.py", line 1643, in add_q
    clause, _ = self._add_q(q_object, can_reuse)
                ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\sql\query.py", line 1675, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ~~~~~~~~~~~~~~~~~^
        child,
        ^^^^^^
    ...<7 lines>...
        update_join_types=update_join_types,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\sql\query.py", line 1523, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
                                        ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\sql\query.py", line 1330, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
                                ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\db\models\sql\query.py", line 1802, in names_to_path
    raise FieldError(
    ...<2 lines>...
    )
django.core.exceptions.FieldError: Cannot resolve keyword 'date_of_joining' into field. Choices are: aadhar_card, admin_projects, approved_leaves, assets, assigned_tasks, bank_details, created_at, current_address, date_of_join, department, department_id, designation, designation_id, dob, documents, educations, email, emergency_contacts, empleavetype, employee_IDs, employee_task, employeechecklist, exit_emails, exitdocument, experiences, failed_login_attempts, gender, handbookacknowledgement, history, history_reported_to, id, is_locked, last_failed_login, leading_projects, leave, message, name, nationality, password, permanent_address, personal_details, phone_number, profile_picture, project_team, reported_to, reported_to_id, resignations, role, status, subordinates, taskhistory, team, timestamp, version, work_location
2025-05-28 18:44:14,010 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report?department=All&month=4&year=2025 HTTP/1.1" 500 122273
2025-05-29 11:31:44,723 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4265, in attendance_report
    return render(request, 'report_templates/Attendence.html', {
        'departments': Department.objects.all(),
    ...<7 lines>...
        'selected_department': selected_department,
    })
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
        contents,
    ...<2 lines>...
        self.engine,
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 976, in do_if
    nodelist = parser.parse(("endif",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 489, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 487, in parse
    filter_expression = self.compile_filter(token.contents)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 609, in compile_filter
    return FilterExpression(token, self)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 705, in __init__
    filter_func = parser.find_filter(filter_name)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 615, in find_filter
    raise TemplateSyntaxError("Invalid filter: '%s'" % filter_name)
django.template.exceptions.TemplateSyntaxError: Invalid filter: 'get_department_name'
2025-05-29 11:31:44,745 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report?department=All&month=5&year=2025 HTTP/1.1" 500 259655
2025-05-29 11:40:24,708 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4265, in attendance_report
    return render(request, 'report_templates/Attendence.html', {
        'departments': Department.objects.all(),
    ...<7 lines>...
        'selected_department': selected_department,
    })
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ~~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 231, in render
    raise ValueError(
    ...<3 lines>...
    )
ValueError: Need 3 values to unpack in for loop; got 2. 
2025-05-29 11:40:24,711 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report?department=All&month=5&year=2025 HTTP/1.1" 500 232221
2025-05-29 11:40:55,811 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4216, in attendance_report
    employees = Employee.objects.all()
                ^^^^^^^^^^^^^^^^
AttributeError: 'function' object has no attribute 'objects'
2025-05-29 11:40:55,814 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report?department=All&month=5&year=2025 HTTP/1.1" 500 68714
2025-05-29 11:41:30,802 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4251, in attendance_report
    return render(request, 'attendance_report.html', context)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: attendance_report.html
2025-05-29 11:41:30,805 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report?department=All&month=5&year=2025 HTTP/1.1" 500 112721
2025-05-29 12:07:16,246 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4257, in attendance_report
    return render(request, 'report_templates/Attendence.html', context)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
        contents,
    ...<2 lines>...
        self.engine,
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 1540, in do_with
    nodelist = parser.parse(("endwith",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 489, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 487, in parse
    filter_expression = self.compile_filter(token.contents)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 609, in compile_filter
    return FilterExpression(token, self)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 705, in __init__
    filter_func = parser.find_filter(filter_name)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 615, in find_filter
    raise TemplateSyntaxError("Invalid filter: '%s'" % filter_name)
django.template.exceptions.TemplateSyntaxError: Invalid filter: 'index'
2025-05-29 12:07:16,257 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report HTTP/1.1" 500 300432
2025-05-29 14:07:38,214 | Level: ERROR | Module: log | Line: 248 | Message: Internal Server Error: /attendence/report
Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\smartif.py", line 180, in translate_token
    op = OPERATORS[token]
         ~~~~~~~~~^^^^^^^
KeyError: 'current_day|add:""|int'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\HR_Project\HR_Portal\HR_Portal\HR_App\views.py", line 4259, in attendance_report
    return render(request, 'report_templates/Attendence.html', context)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
        contents,
    ...<2 lines>...
        self.engine,
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
        (
    ...<2 lines>...
        )
    )
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 961, in do_if
    condition = TemplateIfParser(parser, bits).parse()
                ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 894, in __init__
    super().__init__(*args, **kwargs)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\smartif.py", line 171, in __init__
    mapped_tokens.append(self.translate_token(token))
                         ~~~~~~~~~~~~~~~~~~~~^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\smartif.py", line 182, in translate_token
    return self.create_var(token)
           ~~~~~~~~~~~~~~~^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\defaulttags.py", line 897, in create_var
    return TemplateLiteral(self.template_parser.compile_filter(value), value)
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 609, in compile_filter
    return FilterExpression(token, self)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 705, in __init__
    filter_func = parser.find_filter(filter_name)
  File "C:\HR_Project\HR_Portal\HR_Portal\Lib\site-packages\django\template\base.py", line 615, in find_filter
    raise TemplateSyntaxError("Invalid filter: '%s'" % filter_name)
django.template.exceptions.TemplateSyntaxError: Invalid filter: 'int'
2025-05-29 14:07:38,238 | Level: ERROR | Module: basehttp | Line: 213 | Message: "GET /attendence/report?department=All&month=5&year=2025 HTTP/1.1" 500 339691
